CMakeLists.txt

```diff
@@ -58,6 +58,11 @@ endif()
 
 set (libs)
 
+include(CheckCXXCompilerFlag)
+check_cxx_compiler_flag (-Wformat HAVE_WFORMAT)
+if (HAVE_WFORMAT)
+    set(CMAKE_CXX_FLAGS "${CMAKE_CXX_FLAGS} -Wformat")
```

Comment:
Generally its frowned upon to add global defines. Perhaps we can do this for the affected target(s) using 
[target_compile_definitions](https://cmake.org/cmake/help/latest/command/target_compile_definitions.html) ?

Reply:
But what if we want this flag to apply to all targets? Wouldn't adjusting CMAKE_CXX_FLAGS be appropriate then?

Reply:
Alternatively we would need to add a CMake variable like INPUTLEAP_CXX_FLAGS and pass it to all targets. Looks a bit verbose, no? What are other solutions I'm missing?

Reply:
(I will merge the PR. If we decide this needs changing we can do in a subsequent PR)

Reply:
sure

Reply:
@sithlord48 but please reply to my comments :)

Reply:
IMO compiler *warning* options should always be enable globally (unless there's a really good reason not to) since they 99% of the time catch bugs.

I agree the other compiler options should probably be per-target (though again, depending suse-cases).

thanks heaps for teh quick merge.